geopandas/io/tests/test_parquet.py

```diff
@@ -0,0 +1,231 @@
+
+    filename = os.path.join(str(tmpdir), "test.pq")
+    df.to_parquet(filename)
+
+    assert os.path.exists(filename)
+
+    pq_df = read_parquet(filename)
+
+    assert isinstance(pq_df, GeoDataFrame)
+    assert_geodataframe_equal(df, pq_df, check_like=True)
+
+    assert_geoseries_equal(
+        GeoSeries(df.geom2), GeoSeries(pq_df.geom2), check_geom_type=True
+    )
+
+
+def test_parquet_missing_metadata(tmpdir):
+    """Missing geo metadata, such as from a parquet file created
+    from a pandas DataFrame, will raise a ValueError.
+    """
+
+    test_dataset = "naturalearth_lowres"
+    df = read_file(get_path(test_dataset))
+
+    # convert to DataFrame
+    df = DataFrame(df)
+
+    # convert the geometry column so we can extract later
+    df["geometry"] = to_wkb(df["geometry"].values)
+
+    filename = os.path.join(str(tmpdir), "test.pq")
+
+    # use pandas to_parquet (no geo metadata)
+    df.to_parquet(filename)
+
+    # missing metadata will raise ValueError
+    with pytest.raises(ValueError):
+        read_parquet(filename)
+
+
+def test_parquet_subset_columns(tmpdir):
+    """Reading a subset of columns should correctly decode selected geometry
+    columns.
+    """
+
+    test_dataset = "naturalearth_lowres"
+    df = read_file(get_path(test_dataset))
+
+    filename = os.path.join(str(tmpdir), "test.pq")
+    df.to_parquet(filename)
+    pq_df = read_parquet(filename, columns=["name", "geometry"])
+
+    assert_geodataframe_equal(df[["name", "geometry"]], pq_df, check_like=True)
+
+    with pytest.raises(
+        ValueError, match="No geometry columns are included in the columns read"
+    ):
+        read_parquet(filename, columns=[])
+
+
+def test_parquet_promote_secondary_geometry(tmpdir):
+    """Reading a subset of columns that does not include the primary geometry
+    column should promote the first geometry column present.
+    """
+
+    test_dataset = "naturalearth_lowres"
+    df = read_file(get_path(test_dataset))
+    df["geom2"] = df.geometry.copy()
+
+    filename = os.path.join(str(tmpdir), "test.pq")
+    df.to_parquet(filename)
+    pq_df = read_parquet(filename, columns=["name", "geom2"])
+
+    assert_geodataframe_equal(
+        df.set_geometry("geom2")[["name", "geom2"]], pq_df, check_like=True
+    )
+
+
+def test_parquet_columns_no_geometry(tmpdir):
+    """Reading a parquet file that is missing all of the geometry columns
+    should raise a ValueError"""
+
+    test_dataset = "naturalearth_lowres"
+    df = read_file(get_path(test_dataset))
+
+    filename = os.path.join(str(tmpdir), "test.pq")
+    df.to_parquet(filename)
+
+    with pytest.raises(ValueError):
+        read_parquet(filename, columns=["name"])
+
+
+def test_parquet_missing_crs(tmpdir):
+    """If CRS is `None`, it should be properly handled
+    and remain `None` when read from parquet`.
+    """
+
+    test_dataset = "naturalearth_lowres"
+
+    df = read_file(get_path(test_dataset))
+    df = GeoDataFrame(df, crs=None)
```

Comment:
I am not sure `crs=None` will override the crs of the passed GeoDataFrame (so you might not be testing what you expect) cc @martinfleis, we just discussed a lot of those cornercases in #1339 ;)

I think better to do `df.crs = None`

Reply:
It will not override it. If `df` has CRS, that will be preserved. `crs=None` is default, it cannot override it. `df.crs = None` will do the trick.

Reply:
Good catch!